crates/redis_interface/src/commands.rs

```diff
@@ -174,6 +174,31 @@ impl super::RedisConnectionPool {
             .change_context(errors::RedisError::SetExFailed)
     }
 
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn set_key_if_not_exists_with_expiry<I, O>(
+        &self,
+        key: &str,
+        value: I,
+        seconds: i64,
+    ) -> CustomResult<Option<O>, errors::RedisError>
+    where
+        I: TryInto<RedisValue> + Debug + Send + Sync,
+        I::Error: Into<fred::error::RedisError> + Send + Sync,
+        O: FromRedis,
+    {
+        self.pool
+            .set(
```

Comment:
Set command sets the key regardless if the key exists or not, So not sure how this will be `set_key_if_not_exists` ?